### ISSUE TITLE
Do not add space after commiting completions, but add space to valid commit characters

### ISSUE THREAD

**Issue by DoctorKrolic** [dbeaver/dbeaver#18350]
My preffered style of wrighting sql statements is the folowwing:
```sql
UPDATE table_name
SET ...
```
or
```sql
SELECT column1, column2
FROM ...
```

In both these cases it is convinient to use autocompletion for `table_name` and `column` names. However, dbeaver autocompletion _always_ inserts space after commiting completion. In both cases above this is wrong and annoying. To combine best from both worlds, where you can choose to have a space after commiting, but you are in control of this, IDEs usually have a mechanism of commit characters. For each completion there can be defined a set of characters, pressing which commits completion and it is usually possible for an IDE to have a custom handler for completion commiting for different characters. It wolud be nice if such mechanism is added to the dbeaver.

In particular here I want to have the following behaviour:
- `Tab` and `Enter` commit completion _without_ adding any other chars to it
- `Space` commits completion and adds a space after it.

This scheme is highly flexible, as it gives users the opportunity to choose whether they want to have a space after commiting completions or not. Such behaviour should be implemented for all possible completions, e.g. here is the case, where having a space after keyword is not preffered:
```sql
UPDATE table
SET
  column1=value1
...
```

**Comment by E1izabeth** [comment]
Thank you for feature request.

**Comment by humphreygao** [comment]
- `Tab` and `Enter` commit completion without adding any other chars to it.
You can uncheck this option ->  ![2ZN)I83(BLA_ ALO1%{W}JJ](https://user-images.githubusercontent.com/52394598/204141742-a42453b1-9149-47d5-906b-6d5cd9161dc4.png)

- `Space` commits completion and adds a space after it.
Currently, you can select a code completion option in the code completion popup and press a `Space` to achieve this

**Comment by DoctorKrolic** [comment]
@humphreygao Yeah, but the main point here is that even having such setting it is not flexiable at all. In some cases having a space inserted is really helpful, while in others is is just annoying. My suggestion is to replace this option with the commit characters mechanism I described above

**Comment by humphreygao** [comment]
Insert space :
- [ ] after table names
- [ ] after column names in select clause 
- [ ] after column names in other places

Remove trailing whitespace after pressing `Enter` key

**Comment by DoctorKrolic** [comment]
And then in some dialect we find some edge case, where the expirience is still bad. We add another option for this case, then after a while find another case and add another option. After some time the list would contain 10+ elements and theoretically it still wouldn't cover all possible cases. I don't kwn why you are pushing here towards adding more settings. Having a human to make a desidion is always more flexiable, isn't it?

**Comment by DoctorKrolic** [comment]
Also consider, that even one person might preffer different cases for different scenarios. For instance, I said that I preffer to have
```sql
UPDATE table_name
SET ...
```
but if there is only 1 column to update I might want to wright it like `UPDATE table_name SET column=value...`, and in such case I actually _want_ the space to be inserted when completing `table_name`
